lib/FilterHandler.js

```diff
@@ -3,21 +3,23 @@ import PropTypes from 'prop-types';
 import { filterConfig } from './proptypes';
 import TextFilter from './TextFilter';
 
+var warning = require('warning');
 
-const defaultFilterFunction = ({displayValue, expression}) => {
+const defaultFilterFunction = ({displayValue, expression, columnName}) => {
     if (!(typeof expression === 'string') || expression === ''){
         return true;
     }
     if (displayValue === undefined || displayValue === null){
         return false;
     }
     const valueType = typeof displayValue;
-    if ((valueType === 'string' ) && displayValue.toLowerCase().indexOf(expression.toLowerCase()) === -1){
-        return false;
-    } else if (['number', 'boolean', 'object'].indexOf(valueType) !== -1){
-        return JSON.stringify(displayValue).toLowerCase().indexOf(expression.toLowerCase()) !== -1;
+    if (valueType === 'string'){
+        return displayValue.toLowerCase().indexOf(expression.toLowerCase()) !== -1;
+    }
+    if (process.env.NODE_ENV !== 'production'){
+        warning(false, 'Column "%s" has rendered a value that is not a string. The type of the value (%s) is %s, please specify a custom Filter configuration to be able to filter by this column or disable its filter.', columnName, displayValue, valueType);
     }
-    else return true;
+    return true;
```

Comment:
@fkretzer, maybe we should rather return `false` here so that a string filter never matches a non-string value.

Reply:
No, i think the fallback should be to display the data, instead of hiding it.